### PR TITLE
docs(components): content and markup changes

### PR DESCRIPTION
**Pull request by CITguy**
* Use new basic form control components in demo configs
  * a.k.a., "dogfooding", "drinking our own Kool-aid", etc.
* Normalize Markup
  * Normalize indentation within `{% block %}` definitions
  * Normalize use of `<header>` to wrap section titles and descriptions
* Add TODOs for future documentation updates
  * missing component descriptions
  * missing section descriptions
  * missing table captions
  * etc.
* Normalize nunjucks formatting
  * Use `{{var}}` instead of `{{ var }}` (removed whitespace from within curly braces)
  * Use `{{var}}` instead of `v-text="var"` (snippets should be the only exception)
  * Space around `{% block %}` definitions
  * `{% import %}` before `{% extends %}` before `{% set ... %}`/`{% block %}`

JIRA: SURF-1619

### LGTM's
- [x] Dev LGTM
- [x] Zoom LGTM

### CODE REVIEW

**Comment by HelixUI** [comment]
Deploy preview for *helix-ui* processing.

Building with commit 57607db351bea8dec3c2f50d863e96758dcdce3f

https://app.netlify.com/sites/helix-ui/deploys/5cc7a42640419900074a3f68

File: docs/components/breadcrumbs/index.html [hunk 1]
```diff
@@ -3,20 +3,20 @@
 minver: 0.1.0
 ---
 {% extends 'component.njk' %}
-{% block content %}
-  <hx-alert type="warning" status="WARNING" persist>
-    The last breadcrumb hyperlink is styled to prevent mouse clicks, but it will not prevent keyboard activation.
-  </hx-alert>
```
Comment (CITguy) [inline docs/components/breadcrumbs/index.html]: Moved to new "Accessibility" section at bottom of the page.

File: docs/components/grid/index.html [hunk 2]
```diff
@@ -6,540 +6,557 @@
   components/layouts: Layouts
 ---
 {% extends 'component.njk' %}
-{% block content %}
-<section>
-  <hx-alert type="warning" status="warning" persist>
-    HelixUI grid classes are not compatible with Bootstrap grid styles.
```
Comment (CITguy) [inline docs/components/grid/index.html]: Moved to new "Compatibility" section at bottom of page.

File: docs/components/pagination/index.html [hunk 3]
```diff
@@ -6,15 +6,18 @@
...
+
+{% block page_header %}
+  {# TODO: add component description #}
+{% endblock %}
+
 {% block content %}
   <section>
-    <p>
-      The <code>aria-current</code> attribute needs to be added in order to apply style as well as denote semantic meaning.
-    </p>
```
Comment (CITguy) [inline docs/components/pagination/index.html]: Moved to new "Accessibility" section at bottom of page.

File: docs/components/search/index.html [hunk 4]
```diff
@@ -7,40 +7,68 @@
...
     <div class="example" id="vue-searchDemo" v-cloak>
       <header>
         <form class="beta-hxForm">
-          <p>
-            <label for="txtPlaceholder" class="beta-hxFieldName">Placeholder</label>
+          <hx-text-control>
             <input
-              class="hxTextCtrl"
               id="txtPlaceholder"
-              placeholder="e.g. Search..."
```
Comment (CITguy) [inline docs/components/search/index.html]: placeholders removed to reduce visual noise

File: docs/components/tables/index.html [hunk 5]
```diff
@@ -89,19 +133,23 @@ <h2 id="basic-table">Basic Table</h2>
...
+    </header>
+
     <div class="example">
       <div>
         <table class="hxTable">
           <caption>Most popular Twitter accounts</caption>
           <thead>
             <tr>
               <th class="hxControl">
-                <hx-checkbox></hx-checkbox>
```
Comment (CITguy) [inline docs/components/tables/index.html]: Changed to vanilla checkbox to reduce confusion as we figure out a label-less solution.

File: docs/components/tabs/index.html [hunk 6]
```diff
@@ -11,10 +11,15 @@
 {% set contentClasses = 'docs-tabset' %}
```
Comment (CITguy) [inline docs/components/tabs/index.html]: TODO: place below `extends`

File: docs/docs.less [hunk 7]
```diff
@@ -247,7 +249,7 @@ min-version {
     border-color: @gray-300;
     border-style: solid;
     border-width: 0 0 1px;
-    margin: 0;
+    margin: 0 !important;
```
Comment (CITguy) [inline docs/docs.less]: Removes extra space showing up below example config. `!important` is used, but we'll refactor once we turn the interactive demo into a single-file Vue component.

**Pull request merged**